packages/experiments/src/components/SelectedItemsList/BaseSelectedItemsList.tsx

```diff
@@ -0,0 +1,279 @@
+      key: item.key ? item.key : index,
+      selected: this.selection.isIndexSelected(index),
+      onRemoveItem: () => this.removeItem(item),
+      onItemChange: this.onItemChange,
+      removeButtonAriaLabel: removeButtonAriaLabel,
+      onCopyItem: (itemToCopy: T) => this.copyItems([itemToCopy]),
+    }));
+  }
+
+  protected onSelectionChange(): void {
+    this.forceUpdate();
+  }
+
+  protected onChange(items?: T[]): void {
+    if (this.props.onChange) {
+      // tslint:disable-next-line:no-any
+      (this.props.onChange as any)(items);
+    }
+  }
+
+  @autobind
+  protected onKeyDown(ev: React.KeyboardEvent<HTMLElement>): void {
+    switch (ev.which) {
+      case KeyCodes.backspace:
+        ev.stopPropagation();
+        this.onBackspace(ev);
+        break;
+
+      case KeyCodes.del:
+        this.onBackspace(ev);
+    }
+  }
+
+  @autobind
+  protected onItemChange(changedItem: T, index: number): void {
+    let { items } = this.state;
+
+    if (index >= 0) {
+      let newItems: T[] = items;
+      newItems[index] = changedItem;
+
+      this.updateSelectedItems(newItems);
+    }
+  }
+
+  @autobind
+  protected removeItem(item: ISelectedItemProps<T>): void {
+    let { items } = this.state;
+    let index: number = items.indexOf(item);
+
+    if (index >= 0) {
+      let newItems: T[] = items.slice(0, index).concat(items.slice(index + 1));
+      this.updateSelectedItems(newItems);
+    }
+  }
+
+  @autobind
+  // tslint:disable-next-line:no-any
+  protected removeItems(itemsToRemove: any[]): void {
+    let { items } = this.state;
+    // tslint:disable-next-line:no-any
+    let newItems: T[] = items.filter((item: any) => itemsToRemove.indexOf(item) === -1);
+    let firstItemToRemove = itemsToRemove[0];
+    let index: number = items.indexOf(firstItemToRemove);
+
+    this.updateSelectedItems(newItems, index);
+  }
+
+  // This is protected because we may expect the backspace key to work differently in a different kind of picker.
+  // This lets the subclass override it and provide it's own onBackspace. For an example see the BasePickerListBelow
+  protected onBackspace(ev: React.KeyboardEvent<HTMLElement>): void {
+    if (this.state.items.length) {
+      if (this.selection.getSelectedCount() > 0) {
+        this.removeItems(this.selection.getSelection());
+      } else {
+        this.removeItem(this.state.items[this.state.items.length - 1]);
+      }
+    }
+  }
+
+  /**
+   * Controls what happens whenever there is an action that impacts the selected items.
+   * If selectedItems is provided as a property then this will act as a controlled component and it will not update it's own state.
+  */
+  protected async updateSelectedItems(items: T[], focusIndex?: number): Promise<void> {
+    if (this.props.selectedItems) {
+      // If the component is a controlled component then the controlling component will need
+      this.onChange(items);
+    } else {
+      await this.setState({ items: items }, () => {
```

Comment:
Why is this async and why are you using await? There are no instances where you do updateSelectedItems().then

Reply:
Will remove async and await